middleware/middleware.go

```diff
@@ -38,7 +38,7 @@ func rewriteRulesRegex(rewrite map[string]string) map[*regexp.Regexp]string {
 	rulesRegex := map[*regexp.Regexp]string{}
 	for k, v := range rewrite {
 		k = regexp.QuoteMeta(k)
-		k = strings.Replace(k, `\*`, "(.*)", -1)
+		k = strings.Replace(k, `\*`, "(.*?)", -1)
```

Comment:
This might break existing code using a rewrite like `^/prefix/*/*` which currently capture all path parts for the first capture due to `*` being converted to `(.*)`

Reply:
Will mention it in the important notes (and test against our proxy app)